builder/vsphere/common/step_add_flag.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type FlagConfig
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/template/interpolate"
+	"github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type FlagConfig struct {
+	// Enable Virtualization Based Security option for virtual machine. Defaults to `false`.
+	// Requires `vvtd_enabled` and `NestedHV` to be set to `true`.
+	// Requires `vTPM` to be set to `true`.
+	// Requires `firmware` to be set to `efi-secure`.
+	VbsEnabled bool `mapstructure:"vbs_enabled"`
+	// Enable IO/MMU option for virtual machine. Defaults to `false`.
+	VvtdEnabled bool `mapstructure:"vvtd_enabled"`
+}
+
+func (c *FlagConfig) Prepare(h *HardwareConfig) ([]string, []error) {
+	var errs []error
+	var strings []string
```

Comment:
This strings slice is never populated but returned as an empty slice if there are no errors. I think we can remove it since it is not used for anything.

Reply:
Let me check on that, I know that I needed it for something but will double-check and retest.

Reply:
Addressed in https://github.com/hashicorp/packer-plugin-vsphere/commit/440ec1f12c2d1b30cd6422eb475a8bd8e8807437.